<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2023-05-20" />

<title>cumutlaitve final</title>

<script src="site_libs/header-attrs-2.26/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">dynamicHRpaper</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">cumutlaitve final</h1>
<h4 class="date">2023-05-20</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-06-12
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>dynamicHRpaper/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R
Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240612code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240612)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240612code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240612)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
~/dynamicHRpaper//
</td>
<td style="text-align:left;">
.
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsurbutdynamicHRpapertree9d0696417075ddbada616903c9f8a7b51bad23dftargetblank9d06964a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/surbut/dynamicHRpaper/tree/9d0696417075ddbada616903c9f8a7b51bad23df" target="_blank">9d06964</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomsurbutdynamicHRpapertree9d0696417075ddbada616903c9f8a7b51bad23dftargetblank9d06964a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/surbut/dynamicHRpaper/tree/9d0696417075ddbada616903c9f8a7b51bad23df" target="_blank">9d06964</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Untracked files:
    Untracked:  Figs/
    Untracked:  analysis/.DS_Store
    Untracked:  analysis/Fig1.Rmd
    Untracked:  analysis/Fig1_pcesd.Rmd
    Untracked:  analysis/Fig1b.Rmd
    Untracked:  analysis/Fig2.Rmd
    Untracked:  analysis/Fig4.Rmd
    Untracked:  analysis/Fig5.Rmd
    Untracked:  analysis/Figure3.Rmd
    Untracked:  analysis/GenerationofRawFiles/
    Untracked:  analysis/NRI.Rmd
    Untracked:  analysis/SuppASCVD.Rmd
    Untracked:  analysis/cumulativeinc.Rmd
    Untracked:  analysis/density.Rmd
    Untracked:  analysis/newtables.Rmd
    Untracked:  analysis/zph.Rmd
    Untracked:  code/utils.R
    Untracked:  data/amit_df.rds
    Untracked:  data/amit_df_with_pc_correction.rds
    Untracked:  data/baseline_withPCS.rds
    Untracked:  data/fh_full.rds
    Untracked:  data/fh_prs.rds
    Untracked:  data/fhs_prs_with_pc_corrrection.rds
    Untracked:  data/followupdates.rds
    Untracked:  data/genotype_data_nodup.eigenvec
    Untracked:  output/amit_df_with_pc_correction.rds
    Untracked:  output/fhs_prs_with_pc_corrrection
    Untracked:  output/fhs_prs_with_pc_corrrection.rds
    Untracked:  output/h1hs_withPC.rds
    Untracked:  output/hall_withPC.rds
    Untracked:  output/hallpce.rds
    Untracked:  output/hazards_fh_EQ_withPC_corrections.rds
    Untracked:  output/hazards_fh_EQ_withPC_corrections_sex0.rds
    Untracked:  output/hazards_fh_EQ_withPC_corrections_sex1.rds
    Untracked:  output/hazards_rate_UKB_amit_EQ_withPCcorrection.rds
    Untracked:  output/hazards_rate_lowerbound_amit_EQ_withPCcorrection.rds
    Untracked:  output/hazards_rate_lowerbound_fh_EQ_withPC_corrections.rds
    Untracked:  output/hazards_rate_lowerbound_fh_EQ_withPC_corrections_sex0.rds
    Untracked:  output/hazards_rate_lowerbound_fh_EQ_withPC_corrections_sex1.rds
    Untracked:  output/hazards_rate_upperbound_amit_EQ_withPCcorrection.rds
    Untracked:  output/hazards_rate_upperbound_fh_EQ_withPC_corrections.rds
    Untracked:  output/hazards_rate_upperbound_fh_EQ_withPC_corrections_sex0.rds
    Untracked:  output/hazards_rate_upperbound_fh_EQ_withPC_corrections_sex1.rds
    Untracked:  output/hrindex_fhs_eq_with_pc.csv
    Untracked:  output/hrindex_ukb_eq.csv
    Untracked:  output/r2_withPC.rds
    Untracked:  output/r2fhs_withPC.rds
    Untracked:  output/r2index_fhs_eq_with_pc.csv
    Untracked:  output/r2index_ukb_eq.csv
    Untracked:  output/r2pce.rds

Unstaged changes:
    Modified:   .DS_Store
    Modified:   analysis/about.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="cumulative-incidence-plot" class="section level2">
<h2>Cumulative incidence plot</h2>
</div>
<div id="age-cat-specific-plots" class="section level1">
<h1>Age cat specific plots</h1>
<div id="interaction" class="section level2">
<h2>interaction</h2>
</div>
</div>
<div
id="here-we-break-into-three-age-groups-and-stratify-by-prs-asking-about-resolution-by-prs"
class="section level1">
<h1>Here we break into three age groups and stratify by PRS, asking
about resolution by PRS:</h1>
<pre class="r"><code>dyoung=df[df$phenos.enrollment&lt;55,]
dyoung$pce.age=scale(dyoung$ascvd_10y_accaha_all)
dyoung$pce.r=rank(dyoung$pce.age)/length(dyoung$pce.age)
dyoung$pce.cat=cut(dyoung$pce.r,breaks=c(0,0.20,0.80,1))
dyoung$Interaction=interaction(dyoung$prscat,dyoung$pce.cat)

## now with interaction

fit &lt;- survfit(Surv(phenos.CAD_censor_age,phenos.has_CAD)~Interaction, data=dyoung)
 #fit &lt;- survfit(Surv((phenos.CAD_censor_age), phenos.has_CAD) ~prscat, data=df[df$phenos.enrollment&lt;55,])
g_young=ggsurvplot(fit,
           pval = TRUE, conf.int = TRUE,
        
           fun=&quot;cumhaz&quot;,
           censor=FALSE,
          
           xlim = c(40, 67),
           ylim=c(0,0.6),
           
            #xlab = &quot;Time since Enrollment&quot;,
           xlab = &quot;Time of Event (Age)&quot;,
           ylab = &quot;Cumulative Hazard of Incident CAD&quot;,
           #ylim=c(0,0.15),
           title = &quot;Enrollment Age &lt;55&quot;,
          legend.title = &quot;Genomic and Clinical Risk Strata&quot;,
          ggtheme = theme_classic2(base_size=18),
palette=c(colorRampPalette(c(&quot;#DEEBF7&quot;, &quot;#9ECAE1&quot;, &quot;#3182BD&quot;))(3),
colorRampPalette(c(&quot;#E5F5E0&quot;, &quot;#A1D99B&quot;, &quot;#31A354&quot;))(3),
colorRampPalette(c(&quot;#FEE6CE&quot;, &quot;#FDAE6B&quot;, &quot;#E6550D&quot;))(3)),break.x.by = 5,

          
          legend.labs = c(&quot;Low PCE, Low PRS&quot;,
                          &quot;Low PCE, Intermediate PRS&quot; ,
                          &quot;Low PCE, High PRS&quot;,
                          
                          &quot;Intermediate PCE, Low PRS&quot;,
                          &quot;Intermediate PCE, Intermediate PRS&quot;,
                          &quot;Intermediate PCE, High PRS&quot;,
                          
                          &quot;High PCE, Low PRS&quot;,
                          &quot;High PCE, Intermediate PRS&quot;,
                          &quot;High PCE, High PRS&quot;))

###

dmid=df[df$phenos.enrollment&gt;55&amp;df$phenos.enrollment&lt;65,]
dmid$pce.age=scale(dmid$ascvd_10y_accaha_all)
dmid$pce.r=rank(dmid$pce.age)/length(dmid$pce.age)
dmid$pce.cat=cut(dmid$pce.r,breaks=c(0,0.20,0.80,1))
dmid$Interaction=interaction(dmid$prscat,dmid$pce.cat)
### interaction

fit &lt;- survfit(Surv(phenos.CAD_censor_age,phenos.has_CAD)~Interaction, data=dmid)


g_mid=ggsurvplot(fit,
           pval = TRUE, 
           conf.int = TRUE,
           censor=FALSE,
           ##risk.table = F, 
           ##risk.table.col = &quot;strata&quot;,
           fun=&quot;cumhaz&quot;,
           #xlim = c(0, 15), 
           xlim = c(55, 80),
           ylim=c(0,0.6),
          #xlab = &quot;Time since Enrollment&quot;,
           xlab = &quot;Time of Event (Age)&quot;,
           ylab = &quot;Cumulative Hazard&quot;,
           #ylim=c(0,0.15),
           title = &quot;Enrollment Age 55-65&quot;,
          legend.title = &quot;PRS:PCE Rank&quot;,
          ggtheme = theme_classic2(base_size=18),
           #font.family = &quot;Arial&quot;,
               #  font.family = &quot;Arial&quot;,
          break.x.by = 5,
palette=c(colorRampPalette(c(&quot;#DEEBF7&quot;, &quot;#9ECAE1&quot;, &quot;#3182BD&quot;))(3),
colorRampPalette(c(&quot;#E5F5E0&quot;, &quot;#A1D99B&quot;, &quot;#31A354&quot;))(3),
colorRampPalette(c(&quot;#FEE6CE&quot;, &quot;#FDAE6B&quot;, &quot;#E6550D&quot;))(3)),

          
    
          legend.labs = c(&quot;Low PCE, Low PRS&quot;,
                          &quot;Low PCE, Intermediate PRS&quot; ,
                          &quot;Low PCE, High PRS&quot;,
                          
                          &quot;Intermediate PCE, Low PRS&quot;,
                          &quot;PCE Intermediate, Intermediate PRS&quot;,
                          &quot;Intermediate PCE, High PRS&quot;,
                          
                          &quot;High PCE, Low PRS&quot;,
                          &quot;High PCE, Intermediate PRS&quot;,
                          &quot;High PCE, High PRS&quot;))
###

dold=df[df$phenos.enrollment&gt;65,]
dold$pce.age=scale(dold$ascvd_10y_accaha_all)
dold$pce.r=rank(dold$pce.age)/length(dold$pce.age)
dold$pce.cat=cut(dold$pce.r,breaks=c(0,0.20,0.80,1))
dold$Interaction=interaction(dold$prscat,dold$pce.cat)

### Interaction


fit &lt;- survfit(Surv(phenos.CAD_censor_age, phenos.has_CAD)~Interaction, data=dold) 


g_old=ggsurvplot(fit,
           pval = TRUE, conf.int = TRUE,censor=FALSE,
           ##risk.table = F, 
           ##risk.table.col = &quot;strata&quot;,
           fun=&quot;cumhaz&quot;,
           #xlim = c(0, 15), 
           xlim = c(65, 85),
           ylim=c(0,0.6),
            #xlab = &quot;Time since Enrollment&quot;,
           xlab = &quot;Time of Event (Age)&quot;,
           #ylab = &quot;Cumulative Hazard&quot;,
           #ylim=c(0,0.15),
           title = &quot;Enrollment Age &gt;65&quot;,
          legend.title = &quot;PRS:PCE Rank&quot;,
          ggtheme = theme_classic2(base_size=18),
           #font.family = &quot;Arial&quot;,
           palette=c(colorRampPalette(c(&quot;#DEEBF7&quot;, &quot;#9ECAE1&quot;, &quot;#3182BD&quot;))(3),
colorRampPalette(c(&quot;#E5F5E0&quot;, &quot;#A1D99B&quot;, &quot;#31A354&quot;))(3),
colorRampPalette(c(&quot;#FEE6CE&quot;, &quot;#FDAE6B&quot;, &quot;#E6550D&quot;))(3)),
break.x.by = 5,
       
          legend.labs = c(&quot;Low PCE:Low PRS&quot;,
                          &quot;Low PCE:Intermediate PRS&quot; ,
                          &quot;Low PCE:High PRS&quot;,
                          
                          &quot;Intermediate PCE:Low PRS&quot;,
                          &quot;PCE Intermediate:Intermediate PRS&quot;,
                          &quot;Intermediate PCE:High PRS&quot;,
                          
                          &quot;High PCE:Low PRS&quot;,
                          &quot;High PCE:Intermediate PRS&quot;,
                          &quot;High PCE:High PRS&quot;))</code></pre>
<div id="alltogether" class="section level2">
<h2>alltogether</h2>
<pre class="r"><code>gg1=ggarrange(g_young$plot,g_mid$plot+labs(y=NULL),g_old$plot+labs(y=NULL),nrow=1,ncol=3,labels=c(&quot;A.&quot;,&quot;B.&quot;,&quot;C.&quot;),common.legend = T,legend=&quot;right&quot;)

gg1</code></pre>
<p><img src="figure/Figure3.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggsave(gg1,dpi=300,filename = &quot;Figs/Fig3/interaction_time_byage_withPCS.tiff&quot;,height=10,width = 18,create.dir = TRUE)</code></pre>
<pre><code>✔ Created directory: &#39;Figs/Fig3&#39;.</code></pre>
</div>
<div id="return-cumulative-hazard-for-low-and-high-of-each-age"
class="section level2">
<h2>return cumulative hazard for low and high of each age</h2>
<pre class="r"><code>get_cumI=function(ggsurvplot,stratum){
  stratum_data &lt;- ggsurvplot$data.survplot[ggsurvplot$data.survplot$strata == stratum, ]
  cumulative_hazard &lt;- -log(stratum_data$surv)
  cum_haz_se=stratum_data$std.err/stratum_data$surv
  return(list(ch=cumulative_hazard[length(cumulative_hazard)],se=(stratum_data$std.err/stratum_data$surv)[length(cumulative_hazard)]))
}



get_cumI(g_young,stratum=&quot;Interaction=low.(0,0.2]&quot;)</code></pre>
<pre><code>$ch
[1] 0.004525493

$se
[1] 0.001113627</code></pre>
<pre class="r"><code>get_cumI(g_young,stratum=&quot;Interaction=high.(0.8,1]&quot;)</code></pre>
<pre><code>$ch
[1] 0.1416114

$se
[1] 0.008012507</code></pre>
<pre class="r"><code>get_cumI(g_young,stratum=&quot;Interaction=high.(0.8,1]&quot;)$ch-get_cumI(g_young,stratum=&quot;Interaction=high.(0.8,1]&quot;)$se*1.96</code></pre>
<pre><code>[1] 0.1259069</code></pre>
<pre class="r"><code>get_cumI(g_young,stratum=&quot;Interaction=high.(0.8,1]&quot;)$ch+get_cumI(g_young,stratum=&quot;Interaction=high.(0.8,1]&quot;)$se*1.96</code></pre>
<pre><code>[1] 0.1573159</code></pre>
<pre class="r"><code>get_cumI(g_mid,stratum=&quot;Interaction=low.(0,0.2]&quot;)</code></pre>
<pre><code>$ch
[1] 0.008308347

$se
[1] 0.002048986</code></pre>
<pre class="r"><code>get_cumI(g_mid,stratum=&quot;Interaction=high.(0.8,1]&quot;)</code></pre>
<pre><code>$ch
[1] 0.2710532

$se
[1] 0.07265383</code></pre>
<pre class="r"><code>get_cumI(g_old,stratum=&quot;Interaction=low.(0,0.2]&quot;)</code></pre>
<pre><code>$ch
[1] 0.04624608

$se
[1] 0.02588439</code></pre>
<pre class="r"><code>get_cumI(g_old,stratum=&quot;Interaction=low.(0,0.2]&quot;)$ch-get_cumI(g_old,stratum=&quot;Interaction=low.(0,0.2]&quot;)$se*1</code></pre>
<pre><code>[1] 0.02036169</code></pre>
<pre class="r"><code>get_cumI(g_old,stratum=&quot;Interaction=low.(0,0.2]&quot;)$ch+get_cumI(g_old,stratum=&quot;Interaction=low.(0,0.2]&quot;)$se*1</code></pre>
<pre><code>[1] 0.07213047</code></pre>
<pre class="r"><code>get_cumI(g_old,stratum=&quot;Interaction=low.(0,0.2]&quot;)</code></pre>
<pre><code>$ch
[1] 0.04624608

$se
[1] 0.02588439</code></pre>
<pre class="r"><code>get_cumI(g_old,stratum=&quot;Interaction=high.(0.8,1]&quot;)$ch-get_cumI(g_old,stratum=&quot;Interaction=high.(0.8,1]&quot;)$se*1</code></pre>
<pre><code>[1] 0.1797599</code></pre>
<pre class="r"><code>get_cumI(g_old,stratum=&quot;Interaction=high.(0.8,1]&quot;)$ch+get_cumI(g_old,stratum=&quot;Interaction=high.(0.8,1]&quot;)$se*1</code></pre>
<pre><code>[1] 0.5728429</code></pre>
<pre class="r"><code>get_cumI=function(data,stratum){
  fit &lt;- survfit(Surv(phenos.CAD_censor_age,phenos.has_CAD)~1, data=data[data$Interaction==stratum])
  cumulative_hazard &lt;- fit$cumhaz[length(fit$cumhaz)]
  cum_haz_se=fit$std.err[length(fit$cumhaz)]
  return(list(ch=cumulative_hazard,se=cum_haz_se))
}


get_cumI(dyoung,stratum=&quot;low.(0,0.2]&quot;)</code></pre>
<pre><code>$ch
[1] 0.004524879

$se
[1] 0.001108599</code></pre>
<pre class="r"><code>get_cumI(dyoung,stratum=&quot;high.(0.8,1]&quot;)</code></pre>
<pre><code>$ch
[1] 0.1415855

$se
[1] 0.006954524</code></pre>
<pre class="r"><code>get_cumI(dmid,stratum=&quot;low.(0,0.2]&quot;)</code></pre>
<pre><code>$ch
[1] 0.008306283

$se
[1] 0.002032033</code></pre>
<pre class="r"><code>get_cumI(dmid,stratum=&quot;high.(0.8,1]&quot;)</code></pre>
<pre><code>$ch
[1] 0.2695373

$se
[1] 0.05540406</code></pre>
<pre class="r"><code>get_cumI(dold,stratum=&quot;low.(0,0.2]&quot;)</code></pre>
<pre><code>$ch
[1] 0.04594303

$se
[1] 0.0247146</code></pre>
<pre class="r"><code>get_cumI(dold,stratum=&quot;high.(0.8,1]&quot;)</code></pre>
<pre><code>$ch
[1] 0.3675943

$se
[1] 0.1349052</code></pre>
<pre class="r"><code>## to get standard erro

# https://dominicmagirr.github.io/post/2022-01-18-be-careful-with-standard-errors-in-survival-survfit/

### http://www.sthda.com/english/wiki/survival-analysis-basics


# Get unique levels of the strata variable
strata_levels &lt;- unique(dyoung$Interaction)

# Loop through each level of the strata
for (strata_level in c(&quot;low.(0,0.2]&quot;,&quot;high.(0.8,1]&quot;)) {
  # Subset the survival object for the current strata level
  
strata_surv &lt;- survfit(Surv(phenos.CAD_censor_age,phenos.has_CAD)~1, data=dyoung[dyoung$Interaction==strata_level,])
  
  # Calculate the cumulative hazard
  cumhaz &lt;- -log(strata_surv$surv)[length(strata_surv$surv)]
  lower=-log(strata_surv$upper)[length(strata_surv$surv)]
  upper=-log(strata_surv$lower)[length(strata_surv$surv)]
  

  # Print the results for the current strata level
  cat(&quot;Strata Level:&quot;, strata_level, &quot;\n&quot;)
  cat(&quot;Cumulative Hazard:&quot;, cumhaz, &quot;\n&quot;)
  cat(&quot;Cumulative Hazard Confidence Interval:&quot;, c(lower,upper), &quot;\n\n&quot;)
}</code></pre>
<pre><code>Strata Level: low.(0,0.2] 
Cumulative Hazard: 0.004525493 
Cumulative Hazard Confidence Interval: 0.002352679 0.006698307 

Strata Level: high.(0.8,1] 
Cumulative Hazard: 0.1416114 
Cumulative Hazard Confidence Interval: 0.1279808 0.155242 </code></pre>
<pre class="r"><code># Loop through each level of the strata
for (strata_level in c(&quot;low.(0,0.2]&quot;,&quot;high.(0.8,1]&quot;)) {
  # Subset the survival object for the current strata level
  
strata_surv &lt;- survfit(Surv(phenos.CAD_censor_age,phenos.has_CAD)~1, data=dold[dold$Interaction==strata_level,])
  
  # Calculate the cumulative hazard
  cumhaz &lt;- -log(strata_surv$surv)[length(strata_surv$surv)]
  fit=strata_surv
  lower=fit$cumhaz[length(fit$cumhaz)]-1.96*fit$std.chaz[length(fit$cumhaz)]
  upper=fit$cumhaz[length(fit$cumhaz)]+1.96*fit$std.chaz[length(fit$cumhaz)]
  lower2=-log(strata_surv$upper)[length(strata_surv$surv)]
  upper2=-log(strata_surv$lower)[length(strata_surv$surv)]
  

  # Print the results for the current strata level
  cat(&quot;Strata Level:&quot;, strata_level, &quot;\n&quot;)
  cat(&quot;Cumulative Hazard:&quot;, cumhaz, &quot;\n&quot;)
  cat(&quot;Cumulative Hazard Confidence Interval symmetric:&quot;, c(lower,upper), &quot;\n\n&quot;)
  cat(&quot;Cumulative Hazard Confidence Interval log:&quot;, c(lower2,upper2), &quot;\n\n&quot;)
}</code></pre>
<pre><code>Strata Level: low.(0,0.2] 
Cumulative Hazard: 0.04624608 
Cumulative Hazard Confidence Interval symmetric: -0.001941962 0.09382802 

Cumulative Hazard Confidence Interval log: 0 0.0946858 

Strata Level: high.(0.8,1] 
Cumulative Hazard: 0.3763014 
Cumulative Hazard Confidence Interval symmetric: 0.1199381 0.6152506 

Cumulative Hazard Confidence Interval log: 0.1118921 0.6407107 </code></pre>
<pre class="r"><code>#Among individuals with CAD &lt;55 years, XXX (XXX%) had high (&gt;20th # percentile) PRS and XXX (XXX%) had high (&gt;20% estimated risk) PCE. 

dyoungevent=df[df$phenos.CAD_censor_age&lt;55&amp;df$phenos.has_CAD==1,]
sum(dyoungevent$prs.r&gt;0.80)</code></pre>
<pre><code>[1] 429</code></pre>
<pre class="r"><code>t.test(dyoungevent$prs.r&gt;0.80)</code></pre>
<pre><code>
    One Sample t-test

data:  dyoungevent$prs.r &gt; 0.8
t = 26.625, df = 1084, p-value &lt; 2.2e-16
alternative hypothesis: true mean is not equal to 0
95 percent confidence interval:
 0.3662531 0.4245304
sample estimates:
mean of x 
0.3953917 </code></pre>
<pre class="r"><code>#0.3935484 
sum(dyoungevent$ascvd_10y_accaha_all&gt;20)</code></pre>
<pre><code>[1] 32</code></pre>
<pre class="r"><code>t.test(dyoungevent$ascvd_10y_accaha_all&gt;20)</code></pre>
<pre><code>
    One Sample t-test

data:  dyoungevent$ascvd_10y_accaha_all &gt; 20
t = 5.7395, df = 1084, p-value = 1.232e-08
alternative hypothesis: true mean is not equal to 0
95 percent confidence interval:
 0.01941036 0.03957582
sample estimates:
 mean of x 
0.02949309 </code></pre>
<pre class="r"><code>## 0.02949309</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.0 (2024-04-24)
Platform: aarch64-apple-darwin20
Running under: macOS Sonoma 14.5

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/New_York
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] survminer_0.4.9 ggpubr_0.6.0    dplyr_1.1.4     ggplot2_3.5.1  
[5] survival_3.5-8 

loaded via a namespace (and not attached):
 [1] gtable_0.3.5      xfun_0.44         bslib_0.7.0       rstatix_0.7.2    
 [5] lattice_0.22-6    vctrs_0.6.5       tools_4.4.0       generics_0.1.3   
 [9] tibble_3.2.1      fansi_1.0.6       highr_0.10        pkgconfig_2.0.3  
[13] Matrix_1.7-0      data.table_1.15.4 lifecycle_1.0.4   farver_2.1.2     
[17] compiler_4.4.0    stringr_1.5.1     git2r_0.33.0      textshaping_0.3.7
[21] munsell_0.5.1     carData_3.0-5     httpuv_1.6.15     htmltools_0.5.8.1
[25] sass_0.4.9        yaml_2.3.8        later_1.3.2       pillar_1.9.0     
[29] car_3.1-2         jquerylib_0.1.4   tidyr_1.3.1       cachem_1.1.0     
[33] abind_1.4-5       km.ci_0.5-6       tidyselect_1.2.1  digest_0.6.35    
[37] stringi_1.8.4     purrr_1.0.2       labeling_0.4.3    splines_4.4.0    
[41] cowplot_1.1.3     rprojroot_2.0.4   fastmap_1.2.0     grid_4.4.0       
[45] colorspace_2.1-0  cli_3.6.2         magrittr_2.0.3    utf8_1.2.4       
[49] broom_1.0.6       withr_3.0.0       scales_1.3.0      promises_1.3.0   
[53] backports_1.5.0   rmarkdown_2.26    gridExtra_2.3     ggsignif_0.6.4   
[57] workflowr_1.7.1   ragg_1.3.2        zoo_1.8-12        evaluate_0.23    
[61] knitr_1.46        KMsurv_0.1-5      survMisc_0.5.6    rlang_1.1.4      
[65] Rcpp_1.0.12       xtable_1.8-4      glue_1.7.0        rstudioapi_0.16.0
[69] jsonlite_1.8.8    R6_2.5.1          systemfonts_1.1.0 fs_1.6.4         </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
